src/jit/lower.cpp

```diff
@@ -303,7 +303,7 @@ void Lowering::DecomposeNode(GenTreePtr* pTree, Compiler::fgWalkData* data)
 
             hiStore->gtOp.gtOp1 = hiRhs;
             hiStore->CopyCosts(tree);
-            hiStore->gtFlags |= GTF_VAR_DEF;
+            hiStore->gtFlags = tree->gtFlags;
```

Comment:
I'm not sure this is the proper fix. The issue is that the `hiStore` tree lacks the GTF_EXCEPT flag that overflow checks add and an assert fires. Simply copying all flags appears to work fine.